### ISSUE TITLE
compiling issue with macOS 10.13.5

### ISSUE THREAD

**Issue by thierrygosselin** [curl/curl#2651]
Having problem installing curl-7.60.0 

configure result:
```
./configure --prefix=/usr/local/curl --with-darwinssl
configure: Configured to build curl/libcurl:

  curl version:     7.60.0
  Host setup:       x86_64-apple-darwin17.6.0
  Install prefix:   /usr/local/curl
  Compiler:         gcc
  SSL support:      enabled (Apple OS-native)
  SSH support:      no      (--with-libssh2)
  zlib support:     enabled
  brotli support:   no      (--with-brotli)
  GSS-API support:  no      (--with-gssapi)
  TLS-SRP support:  no      (--enable-tls-srp)
  resolver:         POSIX threaded
  IPv6 support:     enabled
  Unix sockets support: enabled
  IDN support:      no      (--with-{libidn2,winidn})
  Build libcurl:    Shared=yes, Static=yes
  Built-in manual:  enabled
  --libcurl option: enabled (--disable-libcurl-option)
  Verbose errors:   enabled (--disable-verbose)
  SSPI support:     no      (--enable-sspi)
  ca cert bundle:   no
  ca cert path:     
  ca fallback:      
  LDAP support:     enabled (OpenLDAP)
  LDAPS support:    enabled
  RTSP support:     enabled
  RTMP support:     no      (--with-librtmp)
  metalink support: no      (--with-libmetalink)
  PSL support:      no      (libpsl not found)
  HTTP2 support:    disabled (--with-nghttp2)
  Protocols:        DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP
```

make:

```
make -j4
Making all in lib
/Library/Developer/CommandLineTools/usr/bin/make  all-am
  CC       vtls/libcurl_la-darwinssl.lo
In file included from /System/Library/Frameworks/Security.framework/Headers/AuthSession.h:32,
                 from /System/Library/Frameworks/Security.framework/Headers/Security.h:43,
                 from vtls/darwinssl.c:44:
/System/Library/Frameworks/Security.framework/Headers/Authorization.h:193:7: error: variably modified ‘bytes’ at file scope
  char bytes[kAuthorizationExternalFormLength];
       ^~~~~
vtls/darwinssl.c: In function ‘darwinssl_version_from_curl’:
vtls/darwinssl.c:1258:10: error: implicit declaration of function ‘__builtin_available’; did you mean ‘__builtin_scalbl’? [-Werror=implicit-function-declaration]
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
          ^~~~~~~~~~~~~~~~~~~
          __builtin_scalbl
vtls/darwinssl.c:1258:30: error: ‘macOS’ undeclared (first use in this function)
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                              ^~~~~
vtls/darwinssl.c:1258:30: note: each undeclared identifier is reported only once for each function it appears in
vtls/darwinssl.c:1258:35: error: expected ‘)’ before numeric constant
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                                   ^~~~~~
                                   )
vtls/darwinssl.c: In function ‘set_ssl_version_min_max’:
vtls/darwinssl.c:1282:26: error: ‘macOS’ undeclared (first use in this function)
   if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                          ^~~~~
vtls/darwinssl.c:1282:31: error: expected ‘)’ before numeric constant
   if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                               ^~~~~~
                               )
vtls/darwinssl.c: In function ‘darwinssl_connect_step1’:
vtls/darwinssl.c:1434:30: error: ‘macOS’ undeclared (first use in this function)
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                              ^~~~~
vtls/darwinssl.c:1434:35: error: expected ‘)’ before numeric constant
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                                   ^~~~~~
                                   )
vtls/darwinssl.c: In function ‘show_verbose_server_cert’:
vtls/darwinssl.c:2517:6: error: ‘SecTrustEvaluateAsync’ undeclared (first use in this function); did you mean ‘SecTrustEvaluate’?
   if(SecTrustEvaluateAsync != NULL) {
      ^~~~~~~~~~~~~~~~~~~~~
      SecTrustEvaluate
cc1: some warnings being treated as errors
make[2]: *** [vtls/libcurl_la-darwinssl.lo] Error 1
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1
```

any idea how to avoid this ?

**Comment by bagder** [comment]
On my mac mini this does not happen. I tried building both the 7.60.0 release tarball and the current git master just now. Neither shows those compiler errors or notes!

```
$ sw_vers 
ProductName:    Mac OS X
ProductVersion: 10.13.5
BuildVersion:   17F77
$ xcodebuild -version
Xcode 9.4
Build version 9F1027a
$ gcc --version
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**Comment by nickzman** [comment]
If you're using macOS 10.13.x, then make sure you have Xcode 9.x installed. __builtin_available() is an Xcode 9.x addition, and neither Apple nor libcurl support running Xcode 8.x under macOS 10.13.x.

If you do have Xcode 9.x installed, then try running "xcode-select --install" to make sure you have the latest command line developer tools installed.

**Comment by thierrygosselin** [comment]
I have latest command line tools
But I didn’t install Xcode 
Will try

**Comment by thierrygosselin** [comment]
I'm still getting an error (see below)

```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.13.5
BuildVersion:	17F77
```

```
$ xcodebuild -version
Xcode 9.4
Build version 9F1027a
```

[I have the latest GCC for High Sierra](http://hpc.sourceforge.net)
```
$ gcc --version
gcc (GCC) 8.1.0
```

I also have the latest clang
```
$ clang --version
clang version 6.0.0 (tags/RELEASE_600/final)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /usr/local/bin
```

End of `./configure --with-darwinssl`:
```
configure: Configured to build curl/libcurl:

  curl version:     7.60.0
  Host setup:       x86_64-apple-darwin17.6.0
  Install prefix:   /usr/local
  Compiler:         gcc
  SSL support:      enabled (Apple OS-native)
  SSH support:      no      (--with-libssh2)
  zlib support:     enabled
  brotli support:   no      (--with-brotli)
  GSS-API support:  no      (--with-gssapi)
  TLS-SRP support:  no      (--enable-tls-srp)
  resolver:         POSIX threaded
  IPv6 support:     enabled
  Unix sockets support: enabled
  IDN support:      no      (--with-{libidn2,winidn})
  Build libcurl:    Shared=yes, Static=yes
  Built-in manual:  enabled
  --libcurl option: enabled (--disable-libcurl-option)
  Verbose errors:   enabled (--disable-verbose)
  SSPI support:     no      (--enable-sspi)
  ca cert bundle:   no
  ca cert path:     
  ca fallback:      
  LDAP support:     enabled (OpenLDAP)
  LDAPS support:    enabled
  RTSP support:     enabled
  RTMP support:     no      (--with-librtmp)
  metalink support: no      (--with-libmetalink)
  PSL support:      no      (libpsl not found)
  HTTP2 support:    disabled (--with-nghttp2)
  Protocols:        DICT FILE FTP FTPS GOPHER HTTP HTTPS IMAP IMAPS LDAP LDAPS POP3 POP3S RTSP SMB SMBS SMTP SMTPS TELNET TFTP
```

**error during make:**
```
Making all in lib
/Applications/Xcode.app/Contents/Developer/usr/bin/make  all-am
  CC       libcurl_la-file.lo
  CC       libcurl_la-timeval.lo
  CC       libcurl_la-base64.lo
  CC       libcurl_la-hostip.lo
  CC       libcurl_la-progress.lo
  CC       libcurl_la-formdata.lo
  CC       libcurl_la-cookie.lo
  CC       libcurl_la-http.lo
  CC       libcurl_la-sendf.lo
  CC       libcurl_la-ftp.lo
  CC       libcurl_la-url.lo
  CC       libcurl_la-dict.lo
  CC       libcurl_la-if2ip.lo
  CC       libcurl_la-speedcheck.lo
  CC       libcurl_la-ldap.lo
  CC       libcurl_la-version.lo
  CC       libcurl_la-getenv.lo
  CC       libcurl_la-escape.lo
  CC       libcurl_la-mprintf.lo
  CC       libcurl_la-telnet.lo
  CC       libcurl_la-netrc.lo
  CC       libcurl_la-getinfo.lo
  CC       libcurl_la-transfer.lo
  CC       libcurl_la-strcase.lo
  CC       libcurl_la-easy.lo
  CC       libcurl_la-security.lo
  CC       libcurl_la-curl_fnmatch.lo
  CC       libcurl_la-fileinfo.lo
  CC       libcurl_la-ftplistparser.lo
  CC       libcurl_la-wildcard.lo
  CC       libcurl_la-krb5.lo
  CC       libcurl_la-memdebug.lo
  CC       libcurl_la-http_chunks.lo
  CC       libcurl_la-strtok.lo
  CC       libcurl_la-connect.lo
  CC       libcurl_la-llist.lo
  CC       libcurl_la-hash.lo
  CC       libcurl_la-multi.lo
  CC       libcurl_la-content_encoding.lo
  CC       libcurl_la-share.lo
  CC       libcurl_la-http_digest.lo
  CC       libcurl_la-md4.lo
  CC       libcurl_la-md5.lo
  CC       libcurl_la-http_negotiate.lo
  CC       libcurl_la-inet_pton.lo
  CC       libcurl_la-strtoofft.lo
  CC       libcurl_la-strerror.lo
  CC       libcurl_la-amigaos.lo
  CC       libcurl_la-hostasyn.lo
  CC       libcurl_la-hostip4.lo
  CC       libcurl_la-hostip6.lo
  CC       libcurl_la-hostsyn.lo
  CC       libcurl_la-inet_ntop.lo
  CC       libcurl_la-parsedate.lo
  CC       libcurl_la-select.lo
  CC       libcurl_la-tftp.lo
  CC       libcurl_la-splay.lo
  CC       libcurl_la-strdup.lo
  CC       libcurl_la-socks.lo
  CC       libcurl_la-ssh.lo
  CC       libcurl_la-ssh-libssh.lo
  CC       libcurl_la-curl_addrinfo.lo
  CC       libcurl_la-socks_gssapi.lo
  CC       libcurl_la-socks_sspi.lo
  CC       libcurl_la-curl_sspi.lo
  CC       libcurl_la-slist.lo
  CC       libcurl_la-nonblock.lo
  CC       libcurl_la-curl_memrchr.lo
  CC       libcurl_la-imap.lo
  CC       libcurl_la-pop3.lo
  CC       libcurl_la-smtp.lo
  CC       libcurl_la-pingpong.lo
  CC       libcurl_la-rtsp.lo
  CC       libcurl_la-curl_threads.lo
  CC       libcurl_la-warnless.lo
  CC       libcurl_la-hmac.lo
  CC       libcurl_la-curl_rtmp.lo
  CC       libcurl_la-openldap.lo
  CC       libcurl_la-curl_gethostname.lo
  CC       libcurl_la-gopher.lo
  CC       libcurl_la-idn_win32.lo
  CC       libcurl_la-http_proxy.lo
  CC       libcurl_la-non-ascii.lo
  CC       libcurl_la-asyn-ares.lo
  CC       libcurl_la-asyn-thread.lo
  CC       libcurl_la-curl_gssapi.lo
  CC       libcurl_la-http_ntlm.lo
  CC       libcurl_la-curl_ntlm_wb.lo
  CC       libcurl_la-curl_ntlm_core.lo
  CC       libcurl_la-curl_sasl.lo
  CC       libcurl_la-rand.lo
  CC       libcurl_la-curl_multibyte.lo
  CC       libcurl_la-hostcheck.lo
  CC       libcurl_la-conncache.lo
  CC       libcurl_la-pipeline.lo
  CC       libcurl_la-dotdot.lo
  CC       libcurl_la-x509asn1.lo
  CC       libcurl_la-http2.lo
  CC       libcurl_la-smb.lo
  CC       libcurl_la-curl_endian.lo
  CC       libcurl_la-curl_des.lo
  CC       libcurl_la-system_win32.lo
  CC       libcurl_la-mime.lo
  CC       libcurl_la-sha256.lo
  CC       libcurl_la-setopt.lo
  CC       libcurl_la-curl_path.lo
  CC       libcurl_la-curl_ctype.lo
  CC       libcurl_la-curl_range.lo
  CC       vauth/libcurl_la-vauth.lo
  CC       vauth/libcurl_la-cleartext.lo
  CC       vauth/libcurl_la-cram.lo
  CC       vauth/libcurl_la-digest.lo
  CC       vauth/libcurl_la-digest_sspi.lo
  CC       vauth/libcurl_la-krb5_gssapi.lo
  CC       vauth/libcurl_la-krb5_sspi.lo
  CC       vauth/libcurl_la-ntlm.lo
  CC       vauth/libcurl_la-ntlm_sspi.lo
  CC       vauth/libcurl_la-oauth2.lo
  CC       vauth/libcurl_la-spnego_gssapi.lo
  CC       vauth/libcurl_la-spnego_sspi.lo
  CC       vtls/libcurl_la-openssl.lo
  CC       vtls/libcurl_la-gtls.lo
  CC       vtls/libcurl_la-vtls.lo
  CC       vtls/libcurl_la-nss.lo
  CC       vtls/libcurl_la-polarssl.lo
  CC       vtls/libcurl_la-polarssl_threadlock.lo
  CC       vtls/libcurl_la-axtls.lo
  CC       vtls/libcurl_la-cyassl.lo
  CC       vtls/libcurl_la-schannel.lo
  CC       vtls/libcurl_la-schannel_verify.lo
  CC       vtls/libcurl_la-darwinssl.lo
  CC       vtls/libcurl_la-gskit.lo
  CC       vtls/libcurl_la-mbedtls.lo
In file included from /System/Library/Frameworks/Security.framework/Headers/AuthSession.h:32,
                 from /System/Library/Frameworks/Security.framework/Headers/Security.h:43,
                 from vtls/darwinssl.c:44:
/System/Library/Frameworks/Security.framework/Headers/Authorization.h:193:7: error: variably modified ‘bytes’ at file scope
  char bytes[kAuthorizationExternalFormLength];
       ^~~~~
vtls/darwinssl.c: In function ‘darwinssl_version_from_curl’:
vtls/darwinssl.c:1258:10: error: implicit declaration of function ‘__builtin_available’; did you mean ‘__builtin_scalbl’? [-Werror=implicit-function-declaratio]
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
          ^~~~~~~~~~~~~~~~~~~
          __builtin_scalbl
vtls/darwinssl.c:1258:30: error: ‘macOS’ undeclared (first use in this function)
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                              ^~~~~
vtls/darwinssl.c:1258:30: note: each undeclared identifier is reported only once for each function it appears in
vtls/darwinssl.c:1258:35: error: expected ‘)’ before numeric constant
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                                   ^~~~~~
                                   )
vtls/darwinssl.c: In function ‘set_ssl_version_min_max’:
vtls/darwinssl.c:1282:26: error: ‘macOS’ undeclared (first use in this function)
   if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                          ^~~~~
vtls/darwinssl.c:1282:31: error: expected ‘)’ before numeric constant
   if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                               ^~~~~~
                               )
vtls/darwinssl.c: In function ‘darwinssl_connect_step1’:
vtls/darwinssl.c:1434:30: error: ‘macOS’ undeclared (first use in this function)
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                              ^~~~~
vtls/darwinssl.c:1434:35: error: expected ‘)’ before numeric constant
       if(__builtin_available(macOS 10.13, iOS 11.0, *)) {
                                   ^~~~~~
                                   )
vtls/darwinssl.c: In function ‘show_verbose_server_cert’:
vtls/darwinssl.c:2517:6: error: ‘SecTrustEvaluateAsync’ undeclared (first use in this function); did you mean ‘SecTrustEvaluate’?
   if(SecTrustEvaluateAsync != NULL) {
      ^~~~~~~~~~~~~~~~~~~~~
      SecTrustEvaluate
cc1: some warnings being treated as errors
make[2]: *** [vtls/libcurl_la-darwinssl.lo] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [all] Error 2
make: *** [all-recursive] Error 1
```

**Comment by thierrygosselin** [comment]
compiling works fine without ssl:
```
./configure
make -j8
sudo make install
```

but I want ssl...

**Comment by thierrygosselin** [comment]
ok so here what fixed my problem

Re-install OpenSSL
```
wget https://www.openssl.org/source/openssl-1.1.1-pre7.tar.gz
cd openssl-1.1.1-pre7
./config
make
make test
sudo make install
```

Latest curl:
```
./configure --with-ssl
make
sudo make install
```

I also tried after removing Xcode and it still work, so installing Xcode here is optional.

**Comment by bagder** [comment]
That *worked around* your issue since you switched TLS backend, it didn't really solve how to build with securetransport...

**Comment by thierrygosselin** [comment]
exactly, but good enough for me though

**Comment by nickzman** [comment]
Oh, you're building it with GCC instead of Clang? Why?

__builtin_available() is a Clang function, so that's why you're seeing those errors. I never imagined anyone would be using GCC to build the code under High Sierra...

**Comment by nickzman** [comment]
Okay, fair enough. Could you please file a new ticket for me to fix that code so it doesn't assume the user is using Clang as the compiler if they're building it under High Sierra? Thanks.

**Comment by thierrygosselin** [comment]
with clang everything works
```
cd Downloads
wget https://curl.haxx.se/download/curl-7.60.0.tar.gz
tar -zxvf curl-7.60.0.tar.gz
cd curl-7.60.0
CC="/usr/local/bin/clang" ./configure --without-ssl --with-darwinssl
make -j8
sudo make install
```

**Comment by thierrygosselin** [comment]
**new ticket: done**
[issue 2656](https://github.com/curl/curl/issues/2656)
